src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const loadAllGoods = () => {
+    const allGoods = getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+    allGoods.then((data) => {
+      setGoods(data);
+    });
+  };
```

Comment:
1. apply useCallback here
2. don't create variables if you don't really need them. write `getAll().then...`